### DISCUSSION TITLE
Issue using existing OpenSSL in application with other shared libraries

### DISCUSSION THREAD

**Discussion by mrentz-wm** [aws/aws-iot-device-sdk-cpp-v2#547] [Q&A]
**Summary**
When running an application using other shared libraries with the sdk configured with `-DUSE_OPENSSL=ON` I get an error during `s2n_init()`

```
s2n_init() failed: 402653266 (The libcrypto major version number seen at compile-time is different from the major version number seen at run-time) OpenSSL version: 30000020
Fatal error condition occurred in /home/user/projects/aws-workspace/aws-iot-device-sdk-cpp-v2/crt/aws-crt-cpp/crt/aws-c-io/source/s2n/s2n_tls_channel_handler.c:198: 0 && "s2n_init() failed"
Exiting Application
No call stack information available
Aborted (core dumped)
```

I've tried all the options discussed in the [414](https://github.com/aws/aws-iot-device-sdk-cpp-v2/discussions/414) discussion with no success.

The SDK is configured with the following CMake options:
```
cmake -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX="/home/user/projects/aws-workspace" -DUSE_OPENSSL=ON ../aws-iot-device-sdk-cpp-v2
```
CMake and the build finish successfully.

I'm able to build and run a sample app that only uses the SDK but as soon as I add a 3rd party library, I get the above error. 

I added a print message to check the version of OpenSSL loaded. The same version that's printed out during the SDK configure and build process is printed out at run-time.

I checked the other 3rd party library and it doesn't appear to be using OpenSSL or libcrypto but does use libssh2-1 which depends on libcrypt20.

If I build without `-DUSE_OPENSSL=ON` I get the following when building my app:

```
s2n_init() failed: 402653194 (a safety check failed) OpenSSL version: 1010107f
Fatal error condition occurred in /home/user/projects/aws-workspace/aws-iot-device-sdk-cpp-v2/crt/aws-crt-cpp/crt/aws-c-io/source/s2n/s2n_tls_channel_handler.c:198: 0 && "s2n_init() failed"
Exiting Application
No call stack information available
Aborted (core dumped)
```

**Platform**
OS: Linux Mint
CPU: x86_64
Openssl Version 3.0.2
GCC 11.3.0

Any help would be appreciated!

**Comment by mrentz-wm** [comment]
Update:
I am able to successfully use the shared library with the embedded C SDK.

**Reply by tsarquis88** [reply]
> Update: I am able to successfully use the shared library with the embedded C SDK.

Hi @mrentz-wm, I'm facing a similar issue...
Could you expand on how you solved this?

**Comment by mrentz-wm** [comment]
> > Update: I am able to successfully use the shared library with the embedded C SDK.
> 
> Hi @mrentz-wm, I'm facing a similar issue... Could you expand on how you solved this?

I wasn't able to solve the issue. I had to resort to using the embedded C SDK instead in my application.
